src/rotten-tomatoes/index.js

```diff
@@ -1,3 +1,123 @@
-function rottenTomatoes(grid) {}
+function rottenTomatoes(grid) {
+
+    //  Initialize array of adjacents of each tomato
+    let adjacents = [];
+    //  Initialize grid of adjacents according to the grid of tomatoes
+    let adjacentsArr = []
+    //  Initialize queue to use for Breadth First Search
+    let queue = []
+    //  Initialize count to track number of minutes
+    let count = 0
+
+    //  Generate a grid of adjacent positions of each tomato
+    for (let row = 0; row < grid.length; row++){
+        let rowAdjacentArr = []
+        for (let column = 0; column < grid[row].length; column++){
+            adjacents = adjacent(grid.length, grid[row].length, [row, column]);
+            let refinedAdjacents = []
+            for (const position of adjacents){
+                if (position[0] < grid.length && position[1] < grid[0].length){
+                    refinedAdjacents.push(position)
+                }
+            }
+            rowAdjacentArr.push(refinedAdjacents)
+        }
+        adjacentsArr.push(rowAdjacentArr)
+    }
+    
+    //  Locate all rotten tomatoes, queue them at the same level and record their location as visited
+    let visited = []
+    let initialRottenTomatoesAdjacents = []
+    for (let row = 0; row < grid.length; row++){
+        for (let column = 0; column < grid[row].length; column++){
+            if (grid[row][column] === 2){
+                initialRottenTomatoesAdjacents.push(adjacentsArr[row][column])
+                visited.push([row, column])
+            }
+        }
+    }
+    flattened = initialRottenTomatoesAdjacents.flat()
+    queue.push(flattened)
+
+    //  Use Breadth First Search function to count minutes, traverse and update the grid
+    const result = BFS(count, adjacentsArr, queue, grid, visited)
+    count += result[0]
+    grid = result[1]
+
+    //  Return -1 if there is any tomato that will never be rotten ie, shielded/immune tomatoe exist.
+    for (let row = 0; row < grid.length; row++){
+        for (let column = 0; column < grid[row].length; column++){
+            if (grid[row][column] === 1){
+                let immuneAdjacents = 0
+                for (const position of adjacentsArr[row][column]){
+                    if (grid[position[0]][position[1]] === 0 || grid[position[0]][position[1]] === 1){
+                        immuneAdjacents++
+                    }
+                }
+                if (immuneAdjacents === adjacentsArr[row][column].length){
+                    return -1
+                }
+            }
+        }
+    }
+    return count;
+}
+
+// Helpers Functions
+
+//  Function to fetch the adjacent positions of each tomato in the grid
+function adjacent(rowLength, columnLength, position){
+    const top = position[0]===0 ? false : [position[0]-1, position[1]]
```

Comment:
Assigning variables to `position[0]` and `position[1]` makes this way clearer.

Reply:
Noted sir